tests/test_auth.py

```diff
@@ -76,5 +75,15 @@ def route():
             with self.assertRaises(HTTPException) as http_error:
                 route()
 
-if __name__ == '__main__':
-    unittest.main()
+        # Test auth cookie
+        os.environ['SHARED_SECRET'] = 'security'
+        auth = reload(auth)
+
+        with app.test_request_context('/', headers={"COOKIE": "app_secret=security"},
```

Comment:
I suggest wrapping this and the next `with app.test_request_context` within separate `with self.subTest(…):`s, for better test reporting.

Reply:
Thanks for the suggestion, implemented in 4b35300 !